portable/CCS/ARM_CM3/port.c

```diff
@@ -331,6 +331,9 @@ void vPortExitCritical( void )
 
 void xPortSysTickHandler( void )
 {
+    /* The SysTick is not allowed to interrupt a critical section. */
+    configAssert( uxCriticalNesting == 0 )

```

Comment:
We would like to avoid these asserts as these are in very high frequency execution path and therefore, have a negative performance impact. I'd suggest to keep the change limited to the comment update in `list.c`

Reply:
I thought about this but from what I could understand from the documentation configAssert is mostly to be used in development or if you don't have any extreme timing requirements. If you can live with configAssert in other parts of freeRTOS it think this one will have a minor to no impact on the required performance. But as you say I agree that the comment is probably enough to lead people in the right direction when debugging. I'm fine either way.